2020_10_02-traefik-load-balancing/Makefile

```diff
@@ -1,93 +1,115 @@
-	@(ansible --version | head -n 1 | grep -E '(2.9.(1[3-9]|[2-9][0-9])|[3-9].[0-9].[0-9])' > /dev/null) || (echo "packer is not in your PATH or is not up to date" && exit 1)
-	@(which kubectl 2>&1 > /dev/null) || (echo "kubectl is not in your PATH" && exit 1)
-	@$(eval  ANSIBLE_PYTHON_VERSION = $(shell ansible --version | grep 'python version' | awk '{ print $$4 }' | awk -F '.' '{ print $$1 }'))
-	@(which python${ANSIBLE_PYTHON_VERSION} 2>&1 > /dev/null) || (echo "python${ANSIBLE_PYTHON_VERSION} is not in your PATH" && exit 1)
-	@(which pip${ANSIBLE_PYTHON_VERSION} 2>&1 > /dev/null) || (echo "pip${ANSIBLE_PYTHON_VERSION} is not in your PATH" && exit 1)
-	@(pip${ANSIBLE_PYTHON_VERSION} freeze | grep -E 'openshift==0.(1[1-9].[2-9]|[2-9][0-9].[0-9])' > /dev/null) || (echo "openshift>=0.11.2 is not installed as a python${ANSIBLE_PYTHON_VERSION} module" && exit 1)
-	@(pip${ANSIBLE_PYTHON_VERSION} freeze | grep -E 'kubernetes==1[1-9].[0-9].[0-9]' > /dev/null) || (echo "kubernetes>=11.0.0 is not installed as a python${ANSIBLE_PYTHON_VERSION} module" && exit 1)
+	@(ansible --version | head -n 1 | grep -E '(2.9.(1[3-9]|[2-9][0-9])|[3-9].[0-9].[0-9])' > /dev/null) || (echo "ansible is not in your PATH or is not up to date" && exit 1)
+	@(id -u | id -nGz | grep -qzxF libvirt) || (echo "you are not in the libvirt group" && exit 1)
 	@echo "You're ready to play!"
 
-my-linux-env: create-env install-packer install-terraform install-terraform-plugins install-ansible
+my-linux-env: create-env install-packer install-terraform install-terraform-plugins install-ansible ## Install environment.
 
-create-env:
+create-env: ## Create directories and set env variables for binary dependencies.
 	test -d $(LOCAL_BIN)|| mkdir -p $(LOCAL_BIN)
 	echo ${PATH} | grep $(LOCAL_BIN) || (echo 'export PATH=$$PATH:~/.local/bin/' >> ~/.bashrc; . ~/.bashrc)
 
-install-packer:
+install-packer: ## Install Packer.
 	test -f $(LOCAL_BIN)packer || \
 	(curl https://releases.hashicorp.com/packer/$(PACKER_VERSION)/packer_$(PACKER_VERSION)_linux_amd64.zip -o /tmp/packer_$(PACKER_VERSION)_linux_amd64.zip; \
 	unzip /tmp/packer_$(PACKER_VERSION)_linux_amd64.zip -d $(LOCAL_BIN); \
 	rm -f /tmp/packer_$(PACKER_VERSION)_linux_amd64.zip); \
 	chmod +x $(LOCAL_BIN)packer
 
-install-terraform:
+install-terraform: ## Install Terraform.
 	test -f $(LOCAL_BIN)terraform || \
 	(curl -q https://releases.hashicorp.com/terraform/$(TERRAFORM_VERSION)/terraform_$(TERRAFORM_VERSION)_linux_amd64.zip -o /tmp/terraform_$(TERRAFORM_VERSION)_linux_amd64.zip; \
 	unzip /tmp/terraform_$(TERRAFORM_VERSION)_linux_amd64.zip -d $(LOCAL_BIN); \
 	rm -f /tmp/terraform_$(TERRAFORM_VERSION)_linux_amd64.zip); \
 	chmod +x $(LOCAL_BIN)terraform
 
-install-ansible:
+install-ansible: ## Install Ansible with pip.
 	pip3 freeze | grep ansible==$(ANSIBLE_VERSION) || pip3 install ansible kubernetes openshift
```

Comment:
Considering you're now running k8s module on VM directly, is it still necessary to install python modules ?